source/routing/routing.cpp

```diff
@@ -64,10 +64,13 @@ std::string PathItem::print() const {
     return ss.str();
 }
 
+bool is_same_stop_point(const type::EntryPoint& point, const type::StopPoint* stop_point) {
+    return point.type == type::Type_e::StopPoint && point.uri == stop_point->uri;
```

Comment:
just a detail, if stop_point can't be null, I prefer use a ref, if stop_point can be null, it should be checked :)